spacy/tokens/token.pyx

```diff
@@ -350,6 +350,22 @@ cdef class Token:
         def __get__(self):
             return self.c.r_kids
 
+    property sent:
+        """RETURNS (Span): The sentence span that the span is a part of."""
+        def __get__(self):
+            if 'sent' in self.doc.user_span_hooks:
+                return self.doc.user_span_hooks['sent'](self)
+            # This should raise if we're not parsed.
+            self.doc.sents
+            cdef int n = 0
+            root = &self.doc.c[self.i]
+            while root.head != 0:
+                root += root.head
+                n += 1
+                if n >= self.doc.length:
+                    raise RuntimeError
+            return self.doc[root.l_edge:root.r_edge + 1]
+
```

Comment:
Overall I think the delegation approach is probably better here. It's maybe a bit slower, but probably not a material difference. If it's problematic we'd rather refactor. Something like this:

```
def __get__(self):
    if 'sent' in doc.user_token_hooks:
        # I don't really see why this would be useful, but for completeness...
        return doc.user_token_hooks['sent'](self)
     return doc[self.i : self.i+1].sent
```